test/hotspot/jtreg/compiler/lib/compile_framework/Compile.java

```diff
@@ -192,8 +196,10 @@ private static void executeCompileCommand(List<String> command) {
             throw new CompileFrameworkException("InterruptedException during compilation", e);
         }
 
-        if (exitCode != 0 || !output.isEmpty()) {
+        // Note: the output can be non-empty even if the compilation succeeds, e.g. for warnings.
+        if (exitCode != 0) {
```

Comment:
Would `-XX:-PrintWarnings` also work?

Reply:
Maybe... but I don't want to risk it. There have recently been a few sightings where `javac` printed some messages, see [JDK-8351998](https://bugs.openjdk.org/browse/JDK-8351998). I think it's just not worth it to fail if it prints anything. Exit code should be sufficient.

Reply:
Agreed, it's safer.